warehouse/query-core/src/main/java/datawave/query/iterator/logic/TermFrequencyExcerptIterator.java

```diff
@@ -278,6 +300,16 @@ public void next() throws IOException {
                 try {
                     // parse the offsets from the value
                     TermWeight.Info info = TermWeight.Info.parseFrom(source.getTopValue().get());
+                    // check if the number of scores is equal to the number of offsets
+                    boolean useScores = info.getScoreCount() == info.getTermOffsetCount();
+                    boolean hasOnlyNegativeScores = true;
+                    List<Integer> scoreList = info.getScoreList();
+                    for (Integer integer : scoreList) {
+                        if (integer >= 0) {
+                            hasOnlyNegativeScores = false;
+                            break;
+                        }
+                    }
```

Comment:
In future work I would extract this to a helper method

Reply:
done